### DISCUSSION TITLE
Can iconv-lite be installed without using npm?

### DISCUSSION THREAD

**Discussion by irodayo** [community/community#125792] [Programming Help]
### Body

Hello everyone,

I have a question regarding the installation of the iconv-lite package. Typically, I use npm to install Node.js packages, but in my current environment, I am unable to use npm.

Is there a way to manually install iconv-lite without using npm? If so, could you please provide detailed steps on how to achieve this?

Thank you in advance for your help!

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by Abhishek22K** [comment]
To manually install the iconv-lite package without using npm, you can follow these steps:

1. Download the Package
First, download the iconv-lite package from the [npm registry](https://www.npmjs.com/package/iconv-lite).

Go to the iconv-lite page on npm:

[iconv-lite npm page](https://www.npmjs.com/package/iconv-lite)
Download the package as a tarball:

Scroll down to the "Install" section, and find the "Tarball" link. This should be something like https://registry.npmjs.org/iconv-lite/-/iconv-lite-x.y.z.tgz where x.y.z is the version number.
Download the .tgz file to your local machine.
2. Extract the Package
Extract the contents of the downloaded tarball file.

Using a terminal (Command Prompt or PowerShell on Windows, Terminal on macOS/Linux):

tar -xzf iconv-lite-x.y.z.tgz
This will extract the contents into a directory named package.

3. Move the Package to Your Project Directory
Move the extracted directory to the node_modules directory of your project. If node_modules does not exist, create it first.

Create node_modules Directory (if it doesn't exist):

mkdir -p path/to/your/project/node_modules
Move the Package:

mv package path/to/your/project/node_modules/iconv-lite
4. Update Your Project's package.json
If your project has a package.json file, you should add iconv-lite as a dependency manually. This helps keep track of the package and its version.

Open your project's package.json file and add the following dependency (adjust the version number as needed):

{
  "dependencies": {
    "iconv-lite": "x.y.z"
  }
}
If you don’t have a package.json file, you can create one with the minimal structure:

{
  "name": "your-project-name",
  "version": "1.0.0",
  "dependencies": {
    "iconv-lite": "x.y.z"
  }
}
5. Require the Package in Your Code
Finally, you can require and use the iconv-lite package in your Node.js code as usual.

const iconv = require('iconv-lite');

**Comment by yajatkaul** [comment]
yea just use ```npm i iconv-lite```
